esp-hal/src/dma/mod.rs

```diff
@@ -2169,6 +2211,55 @@ where
             }
         }
     }
+
+    /// Configure the channel.
+    pub fn configure(&mut self, burst_mode: bool, priority: DmaPriority) {
+        self.tx.configure(burst_mode, priority);
+        self.rx.configure(burst_mode, priority);
+    }
+
+    /// Converts a blocking channel to an async channel.
+    pub fn into_async(mut self) -> Channel<'d, C, Async> {
+        self.set_interrupt_handler(C::async_handler(&self));
```

Comment:
In general, there should probably be more clean up work when converting to async, since who knows which core the user setup the interrupt handlers on, or if they still have some interrupt pending.

Reply:
Okay I've done some of it but I guess we'll have more edge cases. Since users were able to shoot themselves in the foot previously anyway, I'm fine with the current state - which mostly just neglects peripheral interrupts bits.